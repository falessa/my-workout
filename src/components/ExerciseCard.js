import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#eee',
        padding: 10,
    }
})

const ExerciseCard = ({ exerciseData }) => {
    return (
        <View style={styles.container}>
            <Text>{exerciseData.name}</Text>
            <Text>Sets: {exerciseData.sets}</Text>
            <Text>Reps: {exerciseData.repetitions}</Text>
            <Text>Kgs: {exerciseData.kgs}</Text>
        </View>
        
    )
}

export default ExerciseCard;