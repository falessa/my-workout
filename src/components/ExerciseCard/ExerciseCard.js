import React from 'react';
import { StyleSheet, View, Text, FlatList } from 'react-native';
import { useTranslation } from 'react-i18next';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#eee',
        padding: 10,
        borderRadius: 10
    },
    cardTitle: {
        fontWeight: 'bold',
        fontSize: 17
    },
    boldText: {
        fontWeight: 'bold',
        fontSize: 14
    },
    exerciseDataContainer: {
        flexDirection: 'row',
        paddingTop: 5,
        paddingLeft: 10,
        paddingRight: 10,
        justifyContent: 'center',
        alignContent: 'center'
    }
})

const ExerciseCard = ({ exerciseData }) => {
    const { t } = useTranslation()

    return (
        <View style={styles.container}>
            <Text style={styles.cardTitle}>{exerciseData.name}</Text>
                <View style={styles.exerciseDataContainer}>
                    <FlatList data={exerciseData.setDetails}
                        keyExtractor={item => String(item.id)}
                        renderItem={({ item }) =>
                            <View style={styles.exerciseDataContainer}>
                                <View style={styles.exerciseDataContainer}>
                                    <Text style={styles.boldText}>{item.sets}</Text>
                                    <Text> {t('sets')}</Text>
                                </View>

                                <View style={styles.exerciseDataContainer}>
                                    <Text style={styles.boldText}>{item.repetitions}</Text>
                                    <Text> {t('repetitions')}</Text>
                                </View>

                                <View style={styles.exerciseDataContainer}>
                                    <Text style={styles.boldText}>{item.kgs}</Text>
                                    <Text> {t('kgs')}</Text>
                                </View>
                            </View>
                        } />

                </View>
        </View>
        
    )
}

export default ExerciseCard;