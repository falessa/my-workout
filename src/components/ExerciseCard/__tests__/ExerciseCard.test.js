import "react-native";
import React from "react";
import { create } from "react-test-renderer";
import ExerciseCard from '../ExerciseCard';

describe('ExerciseCard tests', () => {
    const props = {
        exerciseData: {
            name: "Squats",
            sets: "4",
            repetitions: "12",
            kgs: "5"
        }
    }

    const wrapper = create(<ExerciseCard {...props} />)

    test('Render ExerciseCard', () => {
        expect(wrapper).toMatchSnapshot()
    })
})