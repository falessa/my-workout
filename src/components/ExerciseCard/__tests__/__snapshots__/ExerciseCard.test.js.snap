// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseCard tests Render ExerciseCard 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "borderRadius": 10,
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 17,
        "fontWeight": "bold",
      }
    }
  >
    Squats
  </Text>
  <View
    style={
      Object {
        "alignContent": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
  >
    <RCTScrollView
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
